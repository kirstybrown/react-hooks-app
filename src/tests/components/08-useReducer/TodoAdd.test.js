import { shallow } from 'enzyme';

import TodoAdd from "../../../components/08-useReducer/TodoAdd";


describe('Tests for TodoAdd', () => { 

    const handleAddTodo = jest.fn();

    const wrapper = shallow(
        <TodoAdd
            handleAddTodo={ handleAddTodo }
        />
    )

    test('should show component correctly', () => { 

        expect( wrapper ).toMatchSnapshot();
    })

    test('should NOT call handleAddTodo', () => { 

        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });
        expect( handleAddTodo ).toHaveBeenCalledTimes(0);
     })

     test('should call the function handleAddTodo', () => { 
         
        const value = 'Learn React';
        wrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'description'
            }
        });

        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });
        expect( handleAddTodo ).toHaveBeenCalledTimes(1);
        expect( handleAddTodo ).toHaveBeenCalledWith( expect.any(Object) );
        expect( handleAddTodo ).toHaveBeenCalledWith( {
            id: expect.any(Number),
            desc: value,
            done: false
        } );

        expect( wrapper.find('input').prop('value') ).toBe('');

      })
 })