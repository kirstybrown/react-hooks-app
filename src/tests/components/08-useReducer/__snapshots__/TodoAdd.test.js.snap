// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for TodoAdd should show component correctly 1`] = `
<Fragment>
  <h4>
    Add ToDo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Learn..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 btn-block"
      type="submit"
    >
      Add
    </button>
  </form>
</Fragment>
`;
